### PR TITLE
Fix some compiler warnings reported by GCC

### PR DESCRIPTION
**Pull request by Carbenium**
**Changes proposed:**

-  Fix some compiler warnings reported by GCC

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Tests performed:** build

### CODE REVIEW

**Comment by Carbenium** [comment]
An additional ```-Wimplicit-fallthrough``` warning is triggered at https://github.com/TrinityCore/TrinityCore/blob/5af8ff2b84a0c81ce3aa03f338ce838747b2f25e/src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp#L404

It's not 100% sure if that is intended. @Treeston do you have some insight?

File: src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/npc_arthas.cpp [hunk 1]
```diff
@@ -1095,28 +1095,17 @@ class npc_arthas_stratholme : public CreatureScript
...
-                                (*it)->HandleEmoteCommand(emote);
+                                EMOTE_ONESHOT_TALK,
+                                EMOTE_ONESHOT_EXCLAMATION,
+                                EMOTE_ONESHOT_RUDE,
+                                EMOTE_ONESHOT_ROAR
+                            };
+
+                            Creature* victim = Trinity::Containers::SelectRandomContainerElement(nearbyVictims);
+                            if (victim->IsAlive())
+                                victim->HandleEmoteCommand(Trinity::Containers::SelectRandomContainerElement(emotes));
```
Comment (Shauren) [inline src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/npc_arthas.cpp:1108]: `Trinity::Containers::SelectRandomContainerElement` works even on plain c arrays, no need to make this a vector

Comment (Carbenium) [reply src/server/scripts/Kalimdor/CavernsOfTime/CullingOfStratholme/npc_arthas.cpp:1108]: Fixed. Thanks!

**Comment by Treeston** [comment]
Thaddius is a genuine missing break. I think. It's been forever.

But `EVENT_ENGAGE` should've fired just before `EVENT_ENABLE_BALL_LIGHTNING` so react state should already be aggressive there.

(I usually comment my intentional missing breaks too.)

File: src/server/game/Entities/GameObject/GameObject.cpp [hunk 2]
```diff
@@ -649,7 +649,7 @@ void GameObject::Update(uint32 diff)
                     Unit* target = nullptr;
 
                     /// @todo this hack with search required until GO casting not implemented
-                    if (Unit* owner = GetOwner())
```
Comment (Treeston) [inline src/server/game/Entities/GameObject/GameObject.cpp:652]: Can this maybe be done without the objectaccessor lookup?

File: src/server/scripts/Northrend/FrozenHalls/PitOfSaron/instance_pit_of_saron.cpp [hunk 3]
```diff
@@ -157,7 +157,7 @@ class instance_pit_of_saron : public InstanceMapScript
                     case DATA_GARFROST:
                         if (state == DONE)
                         {
-                            if (Creature* summoner = instance->GetCreature(_garfrostGUID))
```
Comment (Treeston) [inline src/server/scripts/Northrend/FrozenHalls/PitOfSaron/instance_pit_of_saron.cpp:160]: Is this check needed for anything? I suspect this is a leftover from Garfrost originally summoned the spawns.

Same for Tyrannus.

Comment (Carbenium) [reply src/server/scripts/Northrend/FrozenHalls/PitOfSaron/instance_pit_of_saron.cpp:160]: You seem to be right. Commit for reference: https://github.com/TrinityCore/TrinityCore/commit/ef694cc251844d34a64ae03f79d28a5c76719f6b

@jackpoz any insight?

File: src/server/game/Entities/GameObject/GameObject.cpp (hunk 2 above)
Comment (Carbenium) [reply src/server/game/Entities/GameObject/GameObject.cpp:652]: No idea. Someone else would have to chime in. @Shauren ?

Comment (Shauren) [reply src/server/game/Entities/GameObject/GameObject.cpp:652]: Comment below makes it suspicious, maybe it was intended to pass `owner` as first arg to `checker` constructor?

Comment (Shauren) [reply src/server/game/Entities/GameObject/GameObject.cpp:652]: Ref 862097a63351e957fd4aa6cb82fe7d9edf4d3926

Comment (Shauren) [reply src/server/game/Entities/GameObject/GameObject.cpp:652]: objectaccessor lookup is fine

File: src/server/scripts/Northrend/FrozenHalls/PitOfSaron/instance_pit_of_saron.cpp (hunk 3 above)
Comment (jackpoz) [reply src/server/scripts/Northrend/FrozenHalls/PitOfSaron/instance_pit_of_saron.cpp:160]: If the summoner npc is not there (whatever is the reason), it makes sense not to spawn the summons. I didn't want to change that logic so I left the null check.

Comment (Carbenium) [reply src/server/scripts/Northrend/FrozenHalls/PitOfSaron/instance_pit_of_saron.cpp:160]: Alright. I'll leave it as it is

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_leotheras_the_blind.cpp [hunk 4]
```diff
@@ -217,7 +217,8 @@ class boss_leotheras_the_blind : public CreatureScript
             IsFinalForm = false;
             NeedThreatReset = false;
             EnrageUsed = false;
-            memset(InnderDemon, 0, sizeof(InnderDemon));
+            for (ObjectGuid& guid : InnderDemon)
+                guid.Clear();
```
Comment (Warpten) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_leotheras_the_blind.cpp:221]: This is redundant,`ObjectGuid` zero-initializes

Comment (Carbenium) [reply src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_leotheras_the_blind.cpp:221]: Correct me if I'm wrong, but isn't `Initialize()` called from `Reset()` as well? The zero-initialization shouldn't help in that case.

Comment (jackpoz) [reply src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_leotheras_the_blind.cpp:221]: it is, this change is required and looks good.

**Comment by jackpoz** [comment]
LGTM

**Review by jackpoz — APPROVED** [review]

**Pull request merged**